templates/master.yaml

```diff
@@ -415,52 +274,91 @@ write_files:
       name: service-account-context
     current-context: service-account-context
-
-- path: /etc/kubernetes/ssl/apiserver-crt.pem.enc
+- path: /opt/bin/decrypt-tls-assets
+  owner: root:root
+  permissions: 0700
+  content: |
+    #!/bin/bash -e
+
+    rkt run \
+      --volume=ssl,kind=host,source=/etc/kubernetes/ssl,readOnly=false \
+      --mount=volume=ssl,target=/etc/kubernetes/ssl \
+      --uuid-file-save=/var/run/coreos/decrypt-tls-assets.uuid \
+      --volume=dns,kind=host,source=/etc/resolv.conf,readOnly=true --mount volume=dns,target=/etc/resolv.conf \
+      --net=host \
+      --trust-keys-from-https \
+      quay.io/coreos/awscli:latest --exec=/bin/bash -- \
+        -ec \
+        'echo decrypting tls assets
+         shopt -s nullglob
+         for encKey in $(find /etc/kubernetes/ssl -name "*.pem.enc"); do
+           echo decrypting $encKey
+           f=$(mktemp $encKey.XXXXXXXX)
+           /usr/bin/aws \
+             --region {{.Spec.Aws.Region}} kms decrypt \
+             --ciphertext-blob fileb://$encKey \
+             --output text \
+             --query Plaintext \
+           | base64 -d > $f
+           mv -f $f ${encKey%.enc}
+         done;
+         echo done.'
+
+    rkt rm --uuid-file=/var/run/coreos/decrypt-tls-assets.uuid || :
+
+- path: /etc/kubernetes/ssl/apiserver.pem.enc
   encoding: gzip+base64
-  content: {{.TLSAssets.APIServerCrt}}
+  content: {{.TLSAssets.APIServerCert}}
 
 - path: /etc/kubernetes/ssl/apiserver-ca.pem.enc
   encoding: gzip+base64
-  content: {{.TLSAssets.APIServerCACrt}}
+  content: {{.TLSAssets.APIServerCACert}}
 
 - path: /etc/kubernetes/ssl/apiserver-key.pem.enc
   encoding: gzip+base64
   content: {{.TLSAssets.APIServerKey}}
 
-- path: /etc/kubernetes/ssl/calico/client-crt.pem.enc
+- path: /etc/kubernetes/ssl/calico/client.pem.enc
   encoding: gzip+base64
-  content: {{.TLSAssets.CalicoClientCrt}}
+  content: {{.TLSAssets.CalicoClientCert}}
 
 - path: /etc/kubernetes/ssl/calico/client-ca.pem.enc
   encoding: gzip+base64
-  content: {{.TLSAssets.CalicoClientCACrt}}
+  content: {{.TLSAssets.CalicoClientCACert}}
 
 - path: /etc/kubernetes/ssl/calico/client-key.pem.enc
   encoding: gzip+base64
   content: {{.TLSAssets.CalicoClientKey}}
 
-- path: /etc/kubernetes/ssl/etcd/server-crt.pem.enc
+- path: /etc/kubernetes/ssl/etcd/server.pem.enc
   encoding: gzip+base64
-  content: {{.TLSAssets.EtcdServerCrt}}
+  content: {{.TLSAssets.EtcdServerCert}}
 
 - path: /etc/kubernetes/ssl/etcd/server-ca.pem.enc
   encoding: gzip+base64
-  content: {{.TLSAssets.EtcdServerCACrt}}
+  content: {{.TLSAssets.EtcdServerCACert}}
 
 - path: /etc/kubernetes/ssl/etcd/server-key.pem.enc
   encoding: gzip+base64
   content: {{.TLSAssets.EtcdServerKey}}
 
-{{range .Files}}- path: {{.Metadata.Path}}
-  owner: {{.Metadata.Owner}}
-  permissions: {{printf "%#o" .Metadata.Permissions}}
-  content: |
-    {{range .Content}}{{.}}
-    {{end}}{{end}}
-
 coreos:
   units:
+  - name: add-route-bridge.service
```

Comment:
that's kvm-specific thing for flannel, should be moved to kvm-operator

summon @rossf7 @xh3b4sd